InterGalacticCurrencyConvertor/src/main/java/materials/MaterialFactory.java

```diff
@@ -0,0 +1,70 @@
+package main.java.materials;
+
+import java.util.HashMap;
+import java.util.Objects;
+
+import main.java.IgnConversionErrors;
+import main.java.IgnConversionException;
+
+/**
+ * Keeps track of different types of materials that can be sold InterGalactically.
+ */
+public final class MaterialFactory {
+  private final HashMap<String, Material> mMtlMap  = new HashMap<>();
+  private final static MaterialFactory mInstance = new MaterialFactory();
+
+  public static MaterialFactory getInstance() {
+    return mInstance;
+  }
+
+  public void add(String name, int numUnits, double totalCost)
+      throws IgnConversionException {
+    Objects.requireNonNull(name, "Invalid Material Name");
+    mMtlMap.put(name.toLowerCase(), new Material(name, numUnits, totalCost));
+  }
+
+  public boolean isAvailable(String name) {
+    return Objects.nonNull(name) && mMtlMap.containsKey(name.toLowerCase());
+  }
+
+  public double getUnitPrice(String name) throws IgnConversionException {
+    Objects.requireNonNull(name, "Invalid Material Name");
+    if (!isAvailable(name)) {
+      throw new IgnConversionException(IgnConversionErrors.INVALID_MATERIAL_MAPPING,
+                                       "Invalid Material " + name + " provided");
```

Comment:
Use String.format(...) instead of string concatenation.

Reply:
done